### PR TITLE
Style: add type hints

### PR DESCRIPTION
**Pull request by ombe1229**
Add some type hints for strict mode

### CODE REVIEW

**Comment by ahopkins** [comment]
Can you take a look at `./sanic/models/handler_types.py`? I think a lot of what you are trying to implement may be there already.

File: sanic/mixins/routes.py [hunk 1]
```diff
@@ -168,7 +169,7 @@ def decorator(handler):
 
     def add_route(
         self,
-        handler: Union[Coroutine[Any, Any, HTTPResponse], HTTPMethodView],
+        handler: MiddlewareResponse,
```
Comment (ahopkins) [inline sanic/mixins/routes.py:172]: Why `MiddlewareResponse`?

File: sanic/views.py [hunk 2]
```diff
@@ -86,7 +86,7 @@ def dispatch_request(self, request, *args, **kwargs):
         return handler(request, *args, **kwargs)
 
     @classmethod
-    def as_view(cls, *class_args, **class_kwargs):
+    def as_view(cls, *class_args: Any, **class_kwargs: Any) -> Any:
```
Comment (ahopkins) [inline sanic/views.py:89]: The return value should be a handler.

File: sanic/mixins/routes.py (hunk 1 above)
Comment (ombe1229) [reply sanic/mixins/routes.py:172]: I missed it. I'll fix that

File: sanic/mixins/listeners.py [hunk 3]
```diff
@@ -1,9 +1,12 @@
 from enum import Enum, auto
 from functools import partial
-from typing import Any, Callable, Coroutine, List, Optional, Union
+from sanic.models.handler_types import SignalHandler
+from typing import Any, Callable, Coroutine, List, Optional, TypeVar, Union
 
 from sanic.models.futures import FutureListener
 
+L = TypeVar("L", bound=SignalHandler)
```
Comment (ahopkins) [inline sanic/mixins/listeners.py:8]: Why `TypeVar` here? I think you meant `ListenerType`. Also, `L` is not a very descriptive variable name.

**Comment by ahopkins** [comment]
I think if we are going to be taking this step, there are probably some more opportunities here. Do you mind if I push a commit to your branch?

**Comment by ombe1229** [comment]
Yes, please. I think there's something I didn't understand exactly.

**Review by ahopkins — APPROVED** [review]

**Pull request merged**